service/matching/pri_backlog_manager.go

```diff
@@ -260,20 +261,31 @@ func (c *priBacklogManagerImpl) BacklogCountHint() (total int64) {
 	return
 }
 
-func (c *priBacklogManagerImpl) BacklogHeadAge() time.Duration {
+func (c *priBacklogManagerImpl) BacklogStatsByPriority() map[int32]*taskqueuepb.TaskQueueStats {
 	c.subqueueLock.Lock()
 	defer c.subqueueLock.Unlock()
 
-	var oldestTime time.Time
-	for _, r := range c.subqueues {
-		oldestTime = minNonZeroTime(oldestTime, r.getOldestBacklogTime())
-	}
-	if oldestTime.IsZero() {
-		// TODO(pri): returning 0 to match existing behavior, but maybe emptyBacklogAge would
-		// be more appropriate in the future.
-		return time.Duration(0)
+	result := make(map[int32]*taskqueuepb.TaskQueueStats)
+	priorityLevels := c.config.PriorityLevels()
+	backlogCounts := c.db.getApproximateBacklogCountsBySubqueue()
+	for idx := range c.subqueues {
+		for priorityIdx := range priorityLevels {
+			priority := priorityIdx + 1 // priority levels start at 1
+			result[priority] = &taskqueuepb.TaskQueueStats{
+				ApproximateBacklogCount: backlogCounts[idx][priorityIdx],
+			}
```

Comment:
how about we just look at the priority keys of the subqueues that exist and aggregate by that (for now there will be only one though)? I don't want to return stats for levels that were never used.